bin/git-info

```diff
@@ -21,27 +36,39 @@ remote_urls() {
 }
 
 echon() {
-    echo "$@"
-    echo
+  echo "$@"
+  echo
 }
 
-# Show info similar to svn
+GREEN="$(tput setaf 2)"
+NORMAL="$(tput sgr0)"
+if [ "$1" = "--color" ] || [ "$2" = "--color" ] || \
+     [ "$1" = "-c" ] || [ "$2" = "-c" ] ; then
+  COLOR_TITLE="$GREEN"
+else
+  COLOR_TITLE="$NORMAL"
+fi
 
 echo
-echon "## Remote URLs:"
+echon "${COLOR_TITLE}## Remote URLs:${NORMAL}"
 echon "$(remote_urls)"
 
-echon "## Remote Branches:"
+echon "${COLOR_TITLE}## Remote Branches:${NORMAL}"
 echon "$(remote_branches)"
 
-echon "## Local Branches:"
+echon "${COLOR_TITLE}## Local Branches:${NORMAL}"
 echon "$(local_branches)"
 
-echon "## Most Recent Commit:"
+SUBMODULES_LOG=$(submodules)
+if [ ! -z "$SUBMODULES_LOG" ]; then
+  echon "${COLOR_TITLE}## Submodule(s):${NORMAL}"
+  echon "$SUBMODULES_LOG"
+fi
+
+echon "${COLOR_TITLE}## Most Recent Commit:${NORMAL}"
 echon "$(most_recent_commit)"
-echon "Type 'git log' for more commits, or 'git show <commit id>' for full commit details."
 
-if test "$1" != "--no-config"; then
-  echon "## Configuration (.git/config):"
+if [ "$1" != "--no-config" ] && [ "$2" != "--no-config" ]; then
```

Comment:
Better to move the argument checks together at the top of the script.

Reply:
Fixed